diff --git a/common.hpp b/common.hpp
index 3789ec8..aeeda98 100644
--- a/common.hpp
+++ b/common.hpp
@@ -72,6 +72,7 @@
 #include <memory>
 #include <algorithm>
 #include <thread>
+#include <cmath>
 
 
 
@@ -153,7 +154,7 @@ public:
 struct Chronometer : Nocopy
 // Requires: no thread is used
 {
-  static constexpr clock_t noclock {-1};
+  static constexpr clock_t noclock {static_cast<clock_t>(-1)};
   static bool enabled;
   const string name;
   clock_t time {0};
