diff -u -ur ncbi-vdb-2.9.6/setup/konfigure.perl ncbi-vdb-2.9.6-fixed/setup/konfigure.perl
--- ncbi-vdb-2.9.6/setup/konfigure.perl	2019-03-18 17:57:05.000000000 +0100
+++ ncbi-vdb-2.9.6-fixed/setup/konfigure.perl	2019-06-24 14:16:55.485625524 +0200
@@ -419,6 +419,7 @@
         exit 1;
     }
 }
+println "now checking JAVAC";
 
 if ($JAVAC) {
     unless (check_tool_h($JAVAC)) {
@@ -426,17 +427,22 @@
         exit 1;
     }
 }
+println "Done checking JAVAC";
 
 my $NO_ARRAY_BOUNDS_WARNING = '';
 if ($TOOLS =~ /gcc$/ && check_no_array_bounds()) {
     $NO_ARRAY_BOUNDS_WARNING = '-Wno-array-bounds';
 }
 
+println "Done checking no-array-bounds";
+
 my $STATIC_LIBSTDCPP = '';
 if ($TOOLS =~ /gcc$/) {
     $STATIC_LIBSTDCPP = check_static_libstdcpp();
 }
 
+println "Done checking static libstdcpp";
+
 if ( $PKG{REQ} ) {
     foreach ( @{ $PKG{REQ} } ) {
         unless (check_tool__h($_)) {
@@ -446,6 +452,8 @@
     }
 }
 
+println "Done blabla";
+
 my @dependencies;
 
 my %DEPEND_OPTIONS;
@@ -1487,6 +1495,7 @@
 }
 
 sub check_no_array_bounds {
+    println "check_no_array_bounds";
     check_compiler('O', '-Wno-array-bounds');
 }
 
